### DISCUSSION TITLE
How to draw gui in two portions

### DISCUSSION THREAD

**Discussion by Jochen0x90h** [ocornut/imgui#3735] [Old/Archived discussions]
Hi!

Is it possible to render the gui in two portions? For example construct a window with ImGui::Begin ... ImGui::End, then obtain the draw data (ImGui::GetDrawData()), draw it, and still in the same frame construct another window with ImGui::Begin ... ImGui::End and then draw it. One use case for this would be taking a screenshot after drawing the first window and before drawing the second window so that only one window is visible in the screenshot. Other use case would be to apply some post-processing effect on only one window.

Thanks,
Jochen

**Comment by ocornut** [comment]
Why not just submitting the first window and not the second when you want to take that screenshot?

As for post-processing, each window will have its own ImDrawList and you can rely on `const char* _OwnerName` or a callback to identify them.

**Reply by Jochen0x90h** [reply]
Thanks. Checking the _OwnerName seems to be what I need and then selectively render stuff (haven't checked the callback method yet).
